src/interval.jl

```diff
@@ -156,6 +156,13 @@ Base.:-(a::Interval{T}, b::T) where T = a + -b
 Base.:-(a::Interval{Char}, b::Integer) = a + -b
 Base.:-(a::Interval{T}, b::Period) where T <: TimeType = a + -b
 
+Base.:-(a::T, b::Interval{T}) where T <: Number = a + -b
+
+function Base.:-(a::Interval{T}) where T <: Number
+    inc = inclusivity(a)
+    Interval{T}(-last(a), -first(a), Inclusivity(last(inc), first(inc)))
+end
```

Comment:
Consider dropping `<: Number` as its possible other types support unary minus outside of numbers.

Reply:
Yeah, I wasn't sure, because I definitely want it to fail for `Char` and `TimeType`, but I guess I can just let it fail on the unary minus itself instead of throwing a `MethodError` here.